docs/spec_draft.md

```diff
@@ -3406,6 +3407,98 @@ More formally, `results[:][result_index] = reduce(input_slices)` where:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reduce_window
+
+### Semantics
+
+Applies a reduction function `body` to windows of `inputs` and `init_values`
+and produces `results`.
+
+The following diagram shows how elements in `results[k]` are computed from
+`inputs[k]` using a concrete example.
+
+<img align="center" src="spec_draft/reduce_window.svg" />
+
+The kth dimension of `inputs[k]` is dilated by placing `base_dilations[k] - 1`
+holes between each of the entries in that dimension.  `padding[k][0]` and
+`padding[k][1]` specify the amount of padding added at the low-end (next to
+index 0) and the high-end (next to the highest index) of dimension `k`
+respectively. The amount of padding can be negative, where the absolute value of
+negative padding indicates the number of elements to remove from the specified
+dimension. Dilation of `inputs[k]` occurs logically before padding, so in the
+case of negative padding, elements are removed from the dilated `inputs[k]`.
```

Comment:
Let's drop this paragraph because: 1) we have an example on the picture above, 2) its contents most duplicate the spec for `pad` which is referenced in the formalism.